test/testutil/provider.c

```diff
@@ -82,12 +82,17 @@ static int fips_provider_version(OSSL_LIB_CTX *libctx, FIPS_VERSION *vers)
     if (!OSSL_PROVIDER_available(libctx, "fips"))
         return 0;
     *params = OSSL_PARAM_construct_utf8_ptr(OSSL_PROV_PARAM_VERSION, &vs, 0);
-    if ((fips_prov = OSSL_PROVIDER_load(libctx, "fips")) == NULL
-            || !OSSL_PROVIDER_get_params(fips_prov, params)
-            || sscanf(vs, "%d.%d.%d", &vers->major, &vers->minor, &vers->patch) != 3
-            || !OSSL_PROVIDER_unload(fips_prov))
+    if ((fips_prov = OSSL_PROVIDER_load(libctx, "fips")) == NULL)
         return -1;
+    if (!OSSL_PROVIDER_get_params(fips_prov, params)
+            || sscanf(vs, "%d.%d.%d", &vers->major, &vers->minor, &vers->patch) != 3)
+        goto err;
+    if (!OSSL_PROVIDER_unload(fips_prov))
+        return -1;  /* WTF do we do here??? */
```

Comment:
I'd just drop the comment :grin:

Reply:
This is on the same level as `close(2)` returning an error.  What to do about it???